src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -158,7 +204,7 @@ def path(self):
         return self._dependency.path
 
     def register_dependency(self, dependency: Dependency) -> DependencyGraph | None:
```

Comment:
```suggestion
    def register_notebook(self, path: Path) -> DependencyGraph | None:
        container = self._resolver.resolve_notebook(self.path, path)
        ...
    
    def register_import(self, module: str) -> DependencyGraph | None:
        container = self._resolver.resolve_import(self.path, module)
        ...
```

we have to be more concrete about the kind of dependency we're registering.